tests/MCPClient/test_policy_check.py

```diff
@@ -0,0 +1,592 @@
+
+
+@pytest.fixture
+def derivation_for_preservation(
+    sip_file: models.File, preservation_file: models.File, event: models.Event
+) -> models.Derivation:
+    return models.Derivation.objects.create(
+        source_file=sip_file, derived_file=preservation_file, event=event
+    )
+
+
+@pytest.fixture
+def preservation_file_format_version(
+    preservation_file: models.File, format_version: fprmodels.FormatVersion
+) -> models.FileFormatVersion:
+    return models.FileFormatVersion.objects.create(
+        file_uuid=preservation_file, format_version=format_version
+    )
+
+
+@pytest.fixture
+def access_file(sip: models.SIP, transfer: models.Transfer) -> models.File:
+    location = b"%SIPDirectory%objects/file.wav"
+    return models.File.objects.create(
+        transfer=transfer,
+        sip=sip,
+        filegrpuse="access",
+        originallocation=location,
+        currentlocation=location,
+    )
+
+
+@pytest.fixture
+def derivation_for_access(
+    sip_file: models.File, access_file: models.File
+) -> models.Derivation:
+    return models.Derivation.objects.create(
+        source_file=sip_file, derived_file=access_file
+    )
+
+
+@pytest.fixture
+def access_file_format_version(
+    access_file: models.File, format_version: fprmodels.FormatVersion
+) -> models.FileFormatVersion:
+    return models.FileFormatVersion.objects.create(
+        file_uuid=access_file, format_version=format_version
+    )
+
+
+STDOUT = json.dumps(
+    {"eventOutcomeInformation": "pass", "eventOutcomeDetailNote": "a note"}
+)
+
+
+@pytest.fixture
+def EXECUTE_OR_RUN() -> Any:
+    with mock.patch(
+        "policy_check.executeOrRun",
+        return_value=(
+            0,
+            STDOUT,
+            "",
+        ),
+    ):
+        yield
+
+
+@pytest.mark.django_db
+@mock.patch("policy_check.executeOrRun")
+def test_policy_check_succeeds_if_rules_exist(
+    execute_or_run: mock.Mock,
+    sip_file: models.File,
+    sip: models.SIP,
+    fprule_policy_check: fprmodels.FPRule,
+    sip_file_format_version: models.FileFormatVersion,
+    format: fprmodels.Format,
+    format_version: fprmodels.FormatVersion,
+    shared_directory_path: str,
+) -> None:
+    job = mock.Mock(
+        args=[
+            "policy_check",
+            sip_file.currentlocation.decode(),
+            str(sip_file.uuid),
+            str(sip.uuid),
+            str(shared_directory_path),
+            sip_file.filegrpuse,
+        ],
+        JobContext=mock.MagicMock(),
+        spec=Job,
+    )
+    expected_stdout = json.dumps(
+        {
+            "eventOutcomeInformation": "pass",
+            "eventOutcomeDetailNote": f'format="{format.description}"; version="{format_version.version}"; result="Well-Formed and valid"',
+        }
+    )
+    execute_or_run.return_value = (0, expected_stdout, "")
+
+    policy_check.call([job])
+
+    job.set_status.assert_called_once_with(0)
```

Comment:
`policy_check` defines constants for its return values:

https://github.com/artefactual/archivematica/blob/01293dc7696ece0066353b6c83ce348ff4c3f3e5/src/MCPClient/lib/clientScripts/policy_check.py#L37-L39

All these `set_status` assertions should use them.

Reply:
ya good point